src/main/java/core/basesyntax/service/operation/PurchaseHandlerImpl.java

```
@@ -0,0 +1,16 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseHandlerImpl implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        if (Storage.fruits.containsKey(fruit)) {
+            Storage.fruits.put(fruit, Storage.fruits.get(fruit) - fruitTransaction.getQuantity());
```

lets throw an exception if there is not enough fruits